lib/index.js

```diff
@@ -0,0 +1,42 @@
+'use strict';
+
+const Bluebird = require('bluebird');
+
+exports.register = (server, options, next) => {
+
+  server.ext('onPreResponse', (request, reply) => {
+
+    const settings = request.route.settings.plugins.totalCount;
+    const credentials = request.auth.credentials;
+    const query = request.query;
+    const filter = request.query.filter || {};
+
+    if (settings && credentials && query.include && query.include.indexOf('total_count') !== -1 && request.response.source) {
+
+      Bluebird.resolve(settings.model)
+      .then((Model) => {
```

Comment:
I know that @mgartner added the `Bluebird.resolve` recently to aid in some New Relic debugging. Apparently `Model.fetch()` only returns a thenable, not a Bluebird promise, so New Relic doesn't instrument it properly.